### PR TITLE
BP 499 edit event query view

### PR DESCRIPTION
**Pull request by tommartensen**

### CODE REVIEW

File: src/CreateEventQueryView/CreateEventQueryView.js [hunk 1]
```diff
@@ -112,24 +212,47 @@ class CreateEventQueryView extends ConnectionComponent {
...
+								onClick={this.submitUpdatedQuery}
 							/>
 						</div>
 					</Container>
 				</div>
 			</div>
 		);
 	}
+
+	componentWillMount() {
```
Comment (MaximilianV) [inline src/CreateEventQueryView/CreateEventQueryView.js]: Please move this function above the custom render methods.

File: src/index.js [hunk 2]
```diff
@@ -20,8 +20,9 @@ ReactDOM.render(
 				}} />
 				<Route path="/grid/:entityId" component={GridView}/>
 				<Route path="/details/:parentId/:entityId" component={DetailView}/>
-			    <Route exact path="/settings" component={SettingsView}/>
-				<Route path="/settings/eventType/:eventTypeId/eventQuery/create" component={CreateEventQueryView}/>
+				<Route exact path="/settings" component={SettingsView}/>
+				<Route exact path="/settings/eventType/:eventTypeId/eventQuery/:isNewQuery/create" component={CreateEventQueryView}/>
+				<Route exact path="/settings/eventType/:eventTypeId/eventQuery/:eventQueryId/:isNewQuery/edit" component={CreateEventQueryView}/>
```
Comment (MaximilianV) [inline src/index.js]: We should consider using an understandable routing. `[...]/eventQuery/false/create` or `[...]/eventQuery/true/create` is not intuitive.
Why do we have to use `:isNewQuery` at all? In the _constructor_ you could check whether `:eventQueryId` is defined or not, couldn't you?

File: src/Utils/RestRoutesManager.js [hunk 3]
```diff
@@ -69,6 +69,14 @@ const availableRoutes = new Map()
 	.set(/^entitymapping\/create$/, {
 		mockFunction: BackendMock.createEntityMapping,
 		isCachable: false,
+	})
+	.set(/^eventquery\/(-?\d+)$/i, {
+		mockFunction: BackendMock.getEventQueryById,
+		isCachable: false
+	})
+    .set(/^eventquery\/(-?\d+)\/edit$/i, {
```
Comment (MaximilianV) [inline src/Utils/RestRoutesManager.js]: Please check indentation in this two new regex definitions.

File: src/Utils/MockData/BackendMock.js [hunk 4]
```diff
@@ -123,5 +127,12 @@ class BackendMock {
 		});
 		return associatedEvents;
 	}
+
+	static getEventQueryById(params) {
+        return Query.find((eventQuery) => {
```
Comment (MaximilianV) [inline src/Utils/MockData/BackendMock.js]: Indentation.

File: src/SettingsView/EventTypeCard.js [hunk 5]
```diff
@@ -141,7 +141,7 @@ class EventType extends ConnectionComponent {
 							<Col md={1}>
 								<IconButton
 									tooltip={<span>create new event query</span>}
-									href={`settings/eventType/${this.props.eventType.Id}/eventQuery/create`}>
+									href={`settings/eventType/${this.props.eventType.Id}/eventQuery/true/create`}>
```
Comment (MaximilianV) [inline src/SettingsView/EventTypeCard.js]: As stated in `index.js`, this is not that nice.

File: src/SettingsView/EventQueryListItem.js [hunk 6]
```diff
@@ -12,26 +12,22 @@ class EventQueryListItem extends Component {
...
+	eventQueryActionButtons () {
 		return (
 			<div>
 				<ConfirmationMessage 
 					actionToPerform={this.deleteEventQuery}
 					ref={(input) => {this.confirmationMessage = input;}}>
 					{config.messages.deleteQueryMessage}
 				</ConfirmationMessage>
-				<IconButton onTouchTap={this.editEventQuery}>
+				<IconButton href={`/settings/eventType/99101991/eventQuery/${this.props.query.Id}/false/edit`}>
```
Comment (MaximilianV) [inline src/SettingsView/EventQueryListItem.js]: Hardcoded ET-Id?

**Review by MaximilianV — CHANGES_REQUESTED** [review]
First comments before tryout.

File: src/CreateEventQueryView/EventQueryInputArea.js [hunk 7]
```diff
@@ -1,27 +1,66 @@
...
 					multiLine={false}
 					fullWidth={true}
-					errorText={this.props.descriptionErrorMessae}
+					errorText={this.props.descriptionErrorMessage}
 					onChange={this.props.handleDescriptionInputChange}
+					defaultValue={loadedDescriptionValue}
 				/>
 				<TextField
-					floatingLabelText={config.descriptions.queryInputFieldHint}
+					floatingLabelText={config.descriptions.queryInputFieldHint + this.getNotEditablePartOfEventQuery()}
```
Comment (MaximilianV) [inline src/CreateEventQueryView/EventQueryInputArea.js:57]: In _real_ create view, this results in `Esper EPL Queryundefined` as floatingLabel.

**Review by MaximilianV — CHANGES_REQUESTED** [review]

File: src/CreateEventQueryView/CreateEventQueryView.js [hunk 8]
```diff
@@ -68,19 +96,19 @@ class CreateEventQueryView extends ConnectionComponent {
...
-			window.history.back();
-			return null;
-		}
-		const eventType = this.props.eventType.value;
-		const attributes = this.props.attributes.value;
-		const connectionIncomplete = super.render(allFetches);
-		if (connectionIncomplete) {
-			return connectionIncomplete;
-		}
+	renderCreateEventQuery() {
```
Comment (MaximilianV) [inline src/CreateEventQueryView/CreateEventQueryView.js]: In `renderCreateEventQuery` and `renderEditEventQuery` there is way too much code duplication.
A parameterized function or splitting it into more, smaller functions might do the trick.

**Review by MaximilianV — CHANGES_REQUESTED** [review]

File: src/Utils/RestRoutesManager.js [hunk 9]
```diff
@@ -74,7 +74,7 @@ const availableRoutes = new Map()
 		mockFunction: BackendMock.getEventQueryById,
 		isCachable: false
 	})
-    .set(/^eventquery\/(-?\d+)\/edit$/i, {
+	.set(/^eventquery\/(-?\d+)\/edit$/i, {
 		mockFunction: BackendMock.updateEventQuery,
         isCachable: false
```
Comment (MaximilianV) [inline src/Utils/RestRoutesManager.js]: Please check indentation again.

**Review by MaximilianV — CHANGES_REQUESTED** [review]

File: src/CreateEventQueryView/EventQueryInputArea.js (hunk 7 above)
Comment (tommartensen) [reply src/CreateEventQueryView/EventQueryInputArea.js:57]: fixed

File: src/SettingsView/EventQueryListItem.js (hunk 6 above)
Comment (tommartensen) [reply src/SettingsView/EventQueryListItem.js]: I thought no one would see this 🤣

**Review by MaximilianV — APPROVED** [review]

**Pull request merged**